lib/shoryuken/queue.rb

```diff
@@ -5,7 +5,11 @@ class Queue
     def initialize(client, name)
       self.name   = name
       self.client = client
-      self.url    = client.get_queue_url(queue_name: name).queue_url
+      begin
+        self.url = client.get_queue_url(queue_name: name).queue_url
+      rescue Aws::SQS::Errors::NonExistentQueue => e
+        raise e.class, "#{e.message} (queue: #{name})", e.backtrace
```

Comment:
Since automatically creating the queue can be too much intrusive, I'd say appending "Did you forget to create this queue?" might be a good aid for starters facing this error.